### PR TITLE
Add k-NN guide and samples

### LINKED ISSUE #558: [FEATURE] Update User guide for knn support (vector database)
### Is your feature request related to a problem?
We recently added support to the java client for using opensearch as a vector database: https://github.com/opensearch-project/opensearch-java/pull/548

### What solution would you like?
We need someone to add an explanation and example code to the user guide.

### LINKED ISSUE #578: [FEATURE] Document A-NN
### Is your feature request related to a problem?

In https://github.com/opensearch-project/opensearch-java/pull/548 we added support for approximate NN. Needs additions to USER_GUIDE of how to use it.

If it helps I added a working sample in https://github.com/dblock/opensearch-java-client-demo/commit/9416501cd8921d03b013e61d792a35c2cc773d3a.

### What solution would you like?

Add a section for approximate nn in USER_GUIDE.

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Documents how to perform approximate k-NN searches and adds complete running samples for the different k-NN queries.

Based off @dblock's work in https://github.com/opensearch-project/opensearch-py/pull/449

### Issues Resolved
Resolves #578
Resolves #558 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -1,21 +1,16 @@
-# User Guide
-
 - [User Guide](#user-guide)
   - [Sample data](#sample-data)
     - [IndexData class](#indexdata-class)
   - [Create a client](#create-a-client)
-    - [Create a client using `RestClientTransport`](#create-a-client-using-restclienttransport)
     - [Create a client using `ApacheHttpClient5Transport`](#create-a-client-using-apachehttpclient5transport)
+    - [Create a client using `RestClientTransport` (deprecated)](#create-a-client-using-restclienttransport-deprecated)
```
Comment (reta) [inline USER_GUIDE.md:6]: thank you

File: samples/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    implementation("org.apache.logging.log4j", "log4j-slf4j2-impl","[2.17.1,3.0)")
+    implementation("commons-logging", "commons-logging", "1.2")
+    implementation("com.fasterxml.jackson.core", "jackson-databind", "2.15.2")
+}
+
+application {
+    mainClass.set("org.opensearch.client.samples.Main")
+}
+
+tasks.named<JavaExec>("run") {
```
Comment (reta) [inline samples/build.gradle.kts:37]: Probably not needed :-)

File: samples/src/main/java/org/opensearch/client/samples/RandUtil.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.samples;
+
+public class RandUtil {
```
Comment (reta) [inline samples/src/main/java/org/opensearch/client/samples/RandUtil.java:11]: Probably `knn` package is better place, I suspect we would have more samples added

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]
Nice work!

**Pull request merged**